src/main/java/com/gdmc/httpinterfacemod/handlers/HeightmapHandler.java

```diff
@@ -0,0 +1,125 @@
+import com.sun.net.httpserver.Headers;
+import com.sun.net.httpserver.HttpExchange;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.world.level.levelgen.Heightmap;
+import net.minecraft.world.level.levelgen.Heightmap.Types;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class HeightmapHandler extends HandlerBase {
+
+    public HeightmapHandler(MinecraftServer mcServer) {
+        super(mcServer);
+    }
+
+
+
+    @Override
+    protected void internalHandle(HttpExchange httpExchange) throws IOException {
+
+        String method = httpExchange.getRequestMethod().toLowerCase();
+
+        if (method.equals("get")) {
+
+            // Get the build area
+            var buildArea = BuildAreaHandler.getBuildArea();
+            if (buildArea == null) {
+                throw new HttpException("No build area is specified. Use the buildarea command inside Minecraft to set a build area.", 404);
+            }
+
+            // Get a reference to the map/level
+            ServerLevel serverlevel = mcServer.overworld();
+
+            // Get query parameters
+            Map<String, String> queryParams = parseQueryString(httpExchange.getRequestURI().getRawQuery());
+            // Try to parse a type argument from them
+            var heightmapTypeString = queryParams.getOrDefault("type", "MOTION_BLOCKING");
+            Types heightmapType;
+            // Check if the type is a valid heightmap type
+            try {
+                // If so, store the type object
+                heightmapType = Types.valueOf(heightmapTypeString);
+            } catch (IllegalArgumentException e) {
+                // Otherwise, throw an error
+                String message = "Could not parse query parameter: " + e.getMessage();
+                throw new HandlerBase.HttpException(message, 400);
+            }
+
+            // Get the heightmap of that type
+            var heightmap = getHeightmap(buildArea, serverlevel, heightmapType);
+
+            // Convert the 2D int array to a 2D JsonArray
+            JsonArray responseArray = new JsonArray();
+            for (int[] row : heightmap) {
+                JsonArray jsonRow = new JsonArray();
+                for (int val : row) {
+                    jsonRow.add(val);
+                }
+                responseArray.add(jsonRow);
+            }
+
+            // Respond with that array as a string
+            Headers responseHeaders = httpExchange.getResponseHeaders();
+            setDefaultResponseHeaders(responseHeaders);
+            resolveRequest(httpExchange, responseArray.toString());
+        }
+        else {
+            throw new HttpException("Method not allowed. Only GET or POST requests are supported.", 405);
+        }
+    }
+
+
+    public int[][] getHeightmap(BuildAreaHandler.BuildArea buildArea, ServerLevel serverlevel, Types heightmapType) {
```

Comment:
Since the final response will be in the form of a `JsonArray` anyway, you might as well construct one inside this method and return that instead of having to convert it afterwards.